packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -512,29 +575,41 @@ const nodeProxyHandler: AdaptingProxyHandler<NodeProxyTarget, EditableTree> = {
                 return target.getField.bind(target);
             case createField:
                 return target.createField.bind(target);
+            case replaceField:
+                return target.replaceField.bind(target);
             default:
                 return undefined;
         }
     },
     set: (
         target: NodeProxyTarget,
         key: string | symbol,
-        value: unknown,
+        value: ContextuallyTypedNodeData,
         receiver: NodeProxyTarget,
     ): boolean => {
         if (typeof key === "string" || symbolIsFieldKey(key)) {
             const fieldKey: FieldKey = brand(key);
-            const fieldKind = target.lookupFieldKind(fieldKey);
-            assert(
-                fieldKind.multiplicity !== Multiplicity.Sequence,
-                0x451 /* Cannot set a value of a sequence field. */,
-            );
-            assert(
-                target.has(fieldKey),
-                0x452 /* The field does not exist. Create the field first using `newFieldSymbol`. */,
-            );
-            const field = target.proxifyField(fieldKey, false);
-            field.getNode(0)[valueSymbol] = value;
+            const fieldSchema = target.getFieldSchema(fieldKey);
+            const multiplicity = target.lookupFieldKind(fieldKey).multiplicity;
+            const content = applyFieldTypesFromContext(target.context.schema, fieldSchema, value);
+            const cursors = content.map(singleMapTreeCursor);
+            if (target.has(fieldKey)) {
```

Comment:
I'm not sure there is a reason to handle these cases differently. Seems like unconditionally using replace field (and making that always take an array) would simplify things here. It might also result in more consistent merging of concurrent edits (I'm not sure if create/replace/delete have the same conflict resolution, but if they don't, using different ones here could be a bit confusing).

Reply:
`replaceField` deletes the field within the transaction. And if the field is not there, it crashes while trying to apply the delta to the forest (I might be wrong in pointing the right place where it crashes).

Reply:
The crash with `replaceField` when the field is not there may be due to a mishandling of 0-length deletes. It would be good to have a repro for that so we can fix it.

I agree with Craig's comment about the merge semantics. Maybe it would be best to first clarify (and document) the merge semantics you want. Ignoring the current implementation and just thinking about the semantics you want long-term, how would you answer the following questions:
- Must a field be populated when `deleteField` is called?
- Must a field be populated when `deleteField` is applied?
- Must a field be populated when `replaceField` is called?
- Must a field be populated when `replaceField` is applied?
- Must a field be empty when `createField` is called?
- Must a field be empty when `createField` is applied?

Reply:
I have added a `TODO` to find out the case when it crashes and to clarify all that.

Reply:
> The crash with `replaceField` when the field is not there may be due to a mishandling of 0-length deletes. It would be good to have a repro for that so we can fix it.

Perhaps, it's already fixed, as I can't reproduce it. So I removed all the code as it was suggested by Craig.